netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -287,16 +305,14 @@ static ProtocolNegotiator createProtocolNegotiator(
 
     private boolean closed;
 
-    private NettyTransportFactory(Class<? extends Channel> channelType,
-                                  NegotiationType negotiationType,
-                                  ProtocolNegotiator protocolNegotiator,
-                                  SslContext sslContext,
-                                  EventLoopGroup group,
-                                  int flowControlWindow,
-                                  int maxMessageSize,
-                                  int maxHeaderListSize) {
+    private NettyTransportFactory(
+        Class<? extends Channel> channelType, Map<ChannelOption<?>, ?> channelOptions,
+        NegotiationType negotiationType, ProtocolNegotiator protocolNegotiator,
+        SslContext sslContext, EventLoopGroup group, int flowControlWindow, int maxMessageSize,
+        int maxHeaderListSize) {
       this.channelType = channelType;
       this.negotiationType = negotiationType;
+      this.channelOptions = new HashMap<ChannelOption<?>, Object>(channelOptions);
```

Comment:
`Collections.unmodifiableMap` only produces RuntimeException when modification is attempted. The declaration of `Map<ChannelOption<?>, ?> channelOptions` already disallows attempts to write code like `channelOptions.put`  because it produces compile time error unless they cast the map to raw type map.